catalog/lib/theme/app_theme.dart

```diff
@@ -102,58 +100,64 @@ class AppTheme {
           filled: true,
           enabledBorder: OutlineInputBorder(
             borderSide: BorderSide(
-              color: _colors.onSurface.shade200,
+              color: colors.onSurface.shade200,
             ),
             borderRadius: BorderRadius.all(Radius.circular(4.r)),
           ),
         ),
       ),
-      primaryColor: _colors.primary,
-      colorScheme: _colors,
       filledButtonTheme: FilledButtonThemeData(
-        style: _buttonStyles.filledButton,
+        style: AppButtonsStyle.getButtonTheme().filledButton,
       ),
       outlinedButtonTheme: OutlinedButtonThemeData(
-        style: _buttonStyles.outlineButton,
+        style: AppButtonsStyle.getButtonTheme().outlineButton,
       ),
       textButtonTheme: TextButtonThemeData(
-        style: _buttonStyles.textButton,
+        style: AppButtonsStyle.getButtonTheme().textButton,
+      ),
+      textTheme: textTheme.apply(
+        bodyColor: CustomColors.getCustomColors().textColor,
+        displayColor: CustomColors.getCustomColors().textColor!.getShade(500),
       ),
-      textTheme: _styles.getThemeData().apply(
-            bodyColor: _colors.textColor,
-            displayColor: _colors.textColor.shade500,
-          ),
       appBarTheme: AppBarTheme(
-        backgroundColor: _colors.primary.shade400,
-        titleTextStyle: TextStyle(color: _colors.textColor.shade100),
+        backgroundColor: colors.primary.shade400,
+        titleTextStyle: TextStyle(
+          color: CustomColors.getCustomColors().textColor!.getShade(500),
+        ),
       ),
+      primaryTextTheme: textTheme,
       checkboxTheme: CheckboxThemeData(
         checkColor: MaterialStateProperty.resolveWith<Color?>(
-          (Set<MaterialState> states) => _colors.primary.shade600,
+          (Set<MaterialState> states) => colors.primary.shade600,
         ),
         fillColor: MaterialStateProperty.resolveWith<Color?>(
-          (Set<MaterialState> states) => _colors.textColor.shade100,
+          (Set<MaterialState> states) => customColors.textColor!.getShade(100),
         ),
         side: BorderSide(
           width: 2,
-          color: _colors.textColor.shade400,
+          color: customColors.textColor!.getShade(400)!,
         ),
       ),
       radioTheme: RadioThemeData(
         fillColor: MaterialStateProperty.resolveWith<Color?>(
-          (Set<MaterialState> states) => _colors.textColor.shade400,
+          (Set<MaterialState> states) => customColors.textColor!.getShade(400),
```

Comment:
Since this is a recurrent "problem", why don't we create a helper function to "resolve" the shades depending on the state. 
In most cases we have the same states enabled, focused, disabled, etc